content/docs/philosophy/philosophy.md

```diff
@@ -0,0 +1,69 @@
+---
+title: Philosophy
+sort_rank: 2
+---
+
+# Do one thing well
+
+We believe in the [Unix
+philosophy](https://en.wikipedia.org/wiki/Unix_philosophy), abridged from [Doug
+McIlroy's initial version from 1978](http://emulator.pdp-11.org.ru/misc/1978.07_-_Bell_System_Technical_Journal.pdf):
+
+1. Make each program do one thing well.
+  While the scope of "one thing" invariably encompasses more and more
+  elements due to increased overall system and computing complexity, we are
+  still doing one thing: ingest metric data, do computations on it, and expose
+  it to other systems.
+2. Expect the output of every program to become the input to another, as yet unknown, program.
+  Today's lingua franca is HTTP endpoints, which are used by Prometheus
+  extensively.
+  In the same vein, Prometheus relies heavily on its own libraries and strict
+  layering internally.
+3. Design and build software, even operating systems, to be tried early, ideally within weeks. Don't hesitate to throw away the clumsy parts and rebuild them.
+  Prometheus will always be available for free as in beer and as in speech.
+  We ensure that master always builds, called Continuous Integration these days,
+  and we not afraid to replace whole sections of our codebase, e.g. our storage
+  engine.
+4. Use tools in preference to unskilled help to lighten a programming task, even if you have to detour to build the tools and expect to throw some of them out after you've finished using them.
+  While this is an outdated way of stating the goal, automation where possible
+  is still one of the core characteristics any modern philosophy.
+
+## Embrace cloud-native technologies
```

Comment:
This is not our philosophy, in fact I'd say this explicitly isn't our philosophy.

We're a technology that happens to work well for things that fall into the "cloud native" marketing term - but we also work just as well outside of that.

Reply:
With that structure, I wanted to say that this works well with good operations, but re-reading, I can see how that got lost over time.

Would you think it better to remove this or to expand on how cloud-native is one of the many facets of proper operations?

Reply:
I'd have a section saying that we work across many paradigms, but aren't going to do things just for the sake of one system that is weird/missing a feature.

Reply:
+1.

Reply:
Rewritten completely & moved down.